metric_tracking/__init__.py

```diff
@@ -0,0 +1,233 @@
+import logging
+import storage
+import mlflow
+
+def do_tracking():
+    available_dates = list(storage.available().keys())
+    available_dates.sort()
+
+    for date_a, date_b in zip(available_dates[:-1], available_dates[1:]):
+        mlflow.start_run()
+        mlflow.log_param("Date A", date_a)
+        mlflow.log_param("Date B", date_b)
+        calculate_score(date_a, date_b)
+        mlflow.end_run()
+
+def calculate_score(date_a, date_b):
+    cluster_a = storage.read(date_a)
+    cluster_b = storage.read(date_b)
+
+    inverse_cluster_a = invert_cluster(cluster_a)
+    inverse_cluster_b = invert_cluster(cluster_b)
+
+    logging.info("calculate score for dates %s - %s" % (date_a, date_b))
+    for cluster_id in inverse_cluster_a.keys():
+        systems_a = frozenset(inverse_cluster_a[cluster_id])
+        systems_b = frozenset(inverse_cluster_b[cluster_id])
+        number_ids_in_both = len(systems_a.intersection(systems_b))
+
+        median_number_ids_in_cluster = (len(systems_a) + len(systems_b)) / 2.0
+        score = number_ids_in_both / median_number_ids_in_cluster
+
+        mlflow.log_metric(("cluster_stability_%s" % cluster_id), score)
+        logging.info("cluster_id %i - stability score %f" % (cluster_id, score))
```

Comment:
right, so score 1 means everything is awesome, 0 nothing is the same, so from the output you've printed the result is pretty bad :-)

1. As you state, the ids of cluster change, so that probably hits the result a lot

2. You've changed the algorithm, before it was focusing only on system_ids present in both runs. 
So now if you have 100 system ids in first and 160 in second (100 are the same and we have 60 new), you'll count the score as 100 / (100+160/2) = 100 / 130 = 0.76. While it should be 1, if the 100 systems were clustered the same. We can't say anything about the 60 new systems (unless we do other analysis of them), so those needs to be ignored.

Reply:
re 2. that's right. good point. I'll add that as a TODO remark